### PR TITLE
small network and security improvement in Dimension Data compute driver

### PR DESCRIPTION
**Pull request by bernard357**
this improvement covers the following:
- fetch ipv6 from node
- allow firewall rule to target a single host address
- allow load balancer to not have a pool

### CODE REVIEW

File: libcloud/compute/drivers/dimensiondata.py [hunk 1]
```diff
@@ -1690,6 +1691,12 @@ def _to_node(self, element):
             if has_network_info else \
             element.find(fixxpath('nic', TYPES_URN)).get('privateIpv4')
 
+        node.extra['ipv6'] = element.find(
+            fixxpath('networkInfo/primaryNic', TYPES_URN)) \
```
Comment (tonybaloney) [inline libcloud/compute/drivers/dimensiondata.py:1699]: You don't need backslashes between brackets.

File: libcloud/loadbalancer/drivers/dimensiondata.py [hunk 2]
```diff
@@ -1041,9 +1041,15 @@ def _to_balancer(self, element):
         port = findtext(element, 'port', TYPES_URN)
         extra = {}
 
-        extra['pool_id'] = element.find(fixxpath(
+        pool_element = element.find(fixxpath(
             'pool',
-            TYPES_URN)).get('id')
+            TYPES_URN))
+        if pool_element is None:
```
Comment (tonybaloney) [inline libcloud/loadbalancer/drivers/dimensiondata.py:1047]: Good idea. I never tested that scenario before

File: libcloud/compute/drivers/dimensiondata.py [hunk 3]
```diff
@@ -1690,6 +1691,12 @@ def _to_node(self, element):
             if has_network_info else \
             element.find(fixxpath('nic', TYPES_URN)).get('privateIpv4')
 
+        node.extra['ipv6'] = element.find(
```
Comment (tonybaloney) [inline libcloud/compute/drivers/dimensiondata.py]: Node is not defined at this point?

Comment (bernard357) [inline libcloud/compute/drivers/dimensiondata.py]: Correct! This should read 
    extra['ipv6'} = ... 

New commit to follow, sorry for the dirty copy and paste.

**Comment by bernard357** [comment]
should be better now - thanks for your feedback

**Comment by tonybaloney** [comment]
Hi Bernard. The tests are still failing. https://travis-ci.org/apache/libcloud/jobs/99012436 
Make sure you run the unit tests locally before committing. Give me a shout if you need help doing that. Another easy thing is to run Travis on your fork

**Comment by bernard357** [comment]
Hello Anthony. Not my fault. Failing tests are related to EC2, as reflected
on the Travis CI page. For the time being I am sticking to the Dimension
Data driver. Thanks for the quick feedback, it helps me to progress on
everything

Bernard

**Comment by tonybaloney** [comment]
that was actually my fault! if you merge the trunk into your branch and recommit the build should be passing now/

**Comment by bernard357** [comment]
Yeah! https://travis-ci.org/apache/libcloud/builds/100236814

**Comment by tonybaloney** [comment]
Ok thanks. Will merge this arvo

**Pull request closed without merging**